renderdoc/driver/gl/gl_hooks_vrapi.cpp

```diff
@@ -0,0 +1,312 @@
+    if(vrapi_GetTextureSwapChainLength_real == NULL)
+      vrapi_GetTextureSwapChainLength_real = (PFN_vrapi_GetTextureSwapChainLength)dlsym(
+          libvrapi_symHandle, "vrapi_GetTextureSwapChainLength");
+    if(vrapi_GetTextureSwapChainHandle_real == NULL)
+      vrapi_GetTextureSwapChainHandle_real = (PFN_vrapi_GetTextureSwapChainHandle)dlsym(
+          libvrapi_symHandle, "vrapi_GetTextureSwapChainHandle");
+    if(vrapi_GetSystemPropertyInt_real == NULL)
+      vrapi_GetSystemPropertyInt_real =
+          (PFN_vrapi_GetSystemPropertyInt)dlsym(libvrapi_symHandle, "vrapi_GetSystemPropertyInt");
+
+    return vrapi_SubmitFrame_real != NULL;
+  }
+
+} vrapi_hooks;
+
+//-----------------------------------------------------------------------------------------------------------------
+GLenum GetInternalFormat(ovrTextureFormat ovr_format)
+{
+  GLenum internalFormat = eGL_RGBA8;
+
+  GLenum conversion_table[] = {
+      eGL_RGBA8,                // VRAPI_TEXTURE_FORMAT_NONE,
+      eGL_RGB565,               // VRAPI_TEXTURE_FORMAT_565,
+      eGL_RGB5_A1,              // VRAPI_TEXTURE_FORMAT_5551,
+      eGL_RGBA4,                // VRAPI_TEXTURE_FORMAT_4444,
+      eGL_RGBA8,                // VRAPI_TEXTURE_FORMAT_8888,
+      eGL_SRGB8_ALPHA8,         // VRAPI_TEXTURE_FORMAT_8888_sRGB,
+      eGL_RGBA16F,              // VRAPI_TEXTURE_FORMAT_RGBA16F,
+      eGL_DEPTH_COMPONENT16,    // VRAPI_TEXTURE_FORMAT_DEPTH_16,
+      eGL_DEPTH_COMPONENT24,    // VRAPI_TEXTURE_FORMAT_DEPTH_24,
+      eGL_DEPTH24_STENCIL8      // VRAPI_TEXTURE_FORMAT_DEPTH_24_STENCIL_8,
+  };
+
+  RDCASSERT(ovr_format < ARRAY_COUNT(conversion_table));
+
+  if(ovr_format < ARRAY_COUNT(conversion_table))
+  {
+    internalFormat = conversion_table[ovr_format];
+  }
+
+  return internalFormat;
+}
+
+GLenum GetTextureType(ovrTextureType ovr_tex_type)
+{
+  GLenum textureType = eGL_TEXTURE_2D;
+
+  GLenum conversion_table[] = {
+      eGL_TEXTURE_2D,          // VRAPI_TEXTURE_TYPE_2D,
+      eGL_TEXTURE_2D,          // VRAPI_TEXTURE_TYPE_2D_EXTERNAL,
+      eGL_TEXTURE_2D_ARRAY,    // VRAPI_TEXTURE_TYPE_2D_ARRAY,
+      eGL_TEXTURE_CUBE_MAP     // VRAPI_TEXTURE_TYPE_CUBE
+  };
+
+  RDCASSERT(ovr_tex_type < ARRAY_COUNT(conversion_table));
+
+  if(ovr_tex_type < ARRAY_COUNT(conversion_table))
+  {
+    textureType = conversion_table[ovr_tex_type];
+  }
+
+  return textureType;
+}
+
+//-----------------------------------------------------------------------------------------------------------------
+extern "C" {
+
+__attribute__((visibility("default"))) ovrMobile *vrapi_EnterVrMode(const ovrModeParms *parms)
+{
+  if(vrapi_hooks.vrapi_EnterVrMode_real == NULL)
+  {
+    vrapi_hooks.SetupHooks();
+  }
+
+  ovrMobile *ovr = vrapi_hooks.vrapi_EnterVrMode_real(parms);
+
+  if(m_GLDriver)
+  {
+    void *ctx = m_GLDriver->GetCtx();
+    void *wndHandle = ovr;
+
+    RenderDoc::Inst().AddFrameCapturer(ctx, wndHandle, m_GLDriver);
+    RenderDoc::Inst().SetActiveWindow(ctx, wndHandle);
```

Comment:
Hmm, so the real problem then is that android doesn't have any way of changing the active window.

There is code in `gl_driver.cpp` to try and handle this complete mess that GL has, with contexts and windows being associated in many-to-many ways. I think this should maybe be fixable in that place. There's some code that iterates through and 'times out' any window that hasn't been presented to with `SwapBuffers` in several seconds, but it doesn't actually unregister the frame capturer.

I'd suggest trying to change that code to unregister the capturer, and then when the capturer for the android window is removed the code will automatically then move the active window to the next available one - the VR handle.